homeassistant/components/zwave_js/discovery_data_template.py

```diff
@@ -107,3 +175,52 @@ def current_temperature_value(resolved_data: dict[str, Any]) -> ZwaveValue | Non
             return lookup_table.get(lookup_key)
 
         return None
+
+
+class NumericSensorDataTemplate(BaseDiscoverySchemaDataTemplate):
+    """Data template class for Z-Wave Sensor entities."""
+
+    def resolve_data(self, value: ZwaveValue) -> dict[str, Any]:
+        """Resolve helper class data for a discovered value."""
+        data = {}
+
+        if value.command_class == CommandClass.BATTERY:
+            data[ATTR_ENTITY_DESC_KEY] = ENTITY_DESC_KEY_BATTERY
+        elif value.command_class == CommandClass.METER:
+            cc_specific = value.metadata.cc_specific
+            meter_type_id = cc_specific[CC_SPECIFIC_METER_TYPE]
+            scale_type_id = cc_specific[CC_SPECIFIC_SCALE]
+            try:
+                meter_type = MeterType(meter_type_id)
+            except ValueError:
+                return data
+            scale_enum = METER_TYPE_TO_SCALE_ENUM_MAP[meter_type]
+            try:
+                scale_type = scale_enum(scale_type_id)
+            except ValueError:
+                return data
+            for key, scale_type_set in METER_DEVICE_CLASS_MAP.items():
+                if scale_type in scale_type_set:
+                    data[ATTR_ENTITY_DESC_KEY] = key
```

Comment:
Maybe break after a hit?
```suggestion
                    data[ATTR_ENTITY_DESC_KEY] = key
                    break
```

Reply:
I'm going to try to make the base data template type more generic, so we can just pass the key directly into the platform data. If that fails, then I will take this approach

Reply:
this worked, so I can just return the key to end the loop